nums/numpy/api/stats.py

```diff
@@ -0,0 +1,1269 @@
+    array([[10,  7,  4],
+           [ 3,  2,  1]])
+    """
+    if axis is not None:
+        raise NotImplementedError("'axis' is currently not supported.")
+    if out is not None:
+        raise NotImplementedError("'out' is currently not supported.")
+    if overwrite_input:
+        raise NotImplementedError("'overwrite_input' is currently not supported.")
+    if interpolation != "linear":
+        raise NotImplementedError(
+            "only 'linear' 'interpolation' is currently supported."
+        )
+    if keepdims:
+        raise NotImplementedError("'keepdims' is currently not supported.")
+    return _instance().percentile(a, q, interpolation=interpolation)
+
+
+def median(a: BlockArray, axis=None, out=None, keepdims=False) -> BlockArray:
+    """Compute the median along the specified axis.
+
+    This docstring was copied from numpy.median.
+
+    Some inconsistencies with the NumS version may exist.
+
+    Returns the median of the array elements.
+
+    Parameters
+    ----------
+    a : BlockArray
+        Input array or object that can be converted to an array.
+    axis : {int, sequence of int, None}, optional
+        Axis or axes along which the medians are computed. The default
+        is to compute the median along a flattened version of the array.
+    out : BlockArray, optional
+        Alternative output array in which to place the result. It must
+        have the same shape and buffer length as the expected output,
+        but the type (of the output) will be cast if necessary.
+    keepdims : bool, optional
+        If this is set to True, the axes which are reduced are left
+        in the result as dimensions with size one. With this option,
+        the result will broadcast correctly against the original `arr`.
+
+    Returns
+    -------
+    median : BlockArray
+        A new array holding the result. If the input contains integers
+        or floats smaller than ``float64``, then the output data-type is
+        ``nps.float64``.  Otherwise, the data-type of the output is the
+        same as that of the input. If `out` is specified, that array is
+        returned instead.
+
+    See Also
+    --------
+    mean, percentile
+
+    Notes
+    -----
+    Given a vector ``V`` of length ``N``, the median of ``V`` is the
+    middle value of a sorted copy of ``V``, ``V_sorted`` - i
+    e., ``V_sorted[(N-1)/2]``, when ``N`` is odd, and the average of the
+    two middle values of ``V_sorted`` when ``N`` is even.
+
+    'axis' is currently not supported.
+
+    'out' is currently not supported.
+
+    'keepdims' is currently not supported.
+    """
+    if axis is not None:
+        raise NotImplementedError("'axis' is currently not supported.")
+    if out is not None:
+        raise NotImplementedError("'out' is currently not supported.")
+    if keepdims:
+        raise NotImplementedError("'keepdims' is currently not supported.")
+    return _instance().median(a)
+
+
+def top_k(
```

Comment:
I would not consider this stat, but not sure where else it would go. @dlzou thoughts?

Reply:
Does moving non-numpy functions to `special.py` sound fine? If our plan is to support specialized operations that NumPy doesn't support, such as adding sparse array and neural network operations, this would be a nice way to isolate it.

Reply:
Sounds good. I refactored them to a new file called `sort.py`. This submodule will contain other routines related to Sort, Search, and Counting to be consistent with here: https://numpy.org/doc/stable/reference/routines.sort.html